dtool/Package.cmake

```diff
@@ -0,0 +1,703 @@
+
+find_package(Python ${WANT_PYTHON_VERSION} QUIET COMPONENTS Interpreter Development)
+
+if(Python_FOUND)
+  set(PYTHON_FOUND ON)
+  set(PYTHON_EXECUTABLE ${Python_EXECUTABLE})
+  set(PYTHON_INCLUDE_DIRS ${Python_INCLUDE_DIRS})
+  set(PYTHON_LIBRARY_DIRS ${Python_LIBRARY_DIRS})
+  set(PYTHON_VERSION_STRING ${Python_VERSION})
+
+else()
+  find_package(PythonInterp ${WANT_PYTHON_VERSION} QUIET)
+  find_package(PythonLibs ${PYTHON_VERSION_STRING} QUIET)
+
+  if(PYTHONLIBS_FOUND)
+    set(PYTHON_FOUND ON)
+
+    if(NOT PYTHON_VERSION_STRING)
+      set(PYTHON_VERSION_STRING ${PYTHONLIBS_VERSION_STRING})
+    endif()
+  endif()
+
+endif()
+
+if(CMAKE_VERSION VERSION_LESS "3.15")
+  # CMake versions this old don't provide Python::Module, so we need to hack up
+  # the variables to ensure no explicit linkage against libpython occurs
+
+  if(WIN32)
+    # Nothing needed here; explicit linkage is appropriate
+    set(PYTHON_LIBRARY "${Python_LIBRARY}")
+    set(PYTHON_LIBRARIES ${Python_LIBRARIES})
+
+  elseif(APPLE OR UNIX)
+    # Just unset and let the implicit linkage take over
+    set(PYTHON_LIBRARY "")
+    set(PYTHON_LIBRARIES "")
+
+    if(APPLE)
+      # macOS requires this explicit flag on the linker command line to allow the
+      # references to the Python symbols to resolve at dynamic link time
+      string(APPEND CMAKE_MODULE_LINKER_FLAGS " -undefined dynamic_lookup")
+      # TODO: p3dcparser contains some direct Python references; get rid of
+      # this once that's gone
+      string(APPEND CMAKE_SHARED_LINKER_FLAGS " -undefined dynamic_lookup")
+
+    endif()
+
+  else()
+    # On every other platform, guessing is a bad idea - insist the user upgrade
+    # their CMake instead.
+    message(WARNING "For Python support on this platform, please use CMake >= 3.15!")
+    set(PYTHON_FOUND OFF)
+
+  endif()
+
+endif()
+
+package_option(PYTHON
+  DEFAULT ON
+  "Enables support for Python.  If INTERROGATE_PYTHON_INTERFACE
+is also enabled, Python bindings will be generated."
+  IMPORTED_AS Python::Module)
+
+# Also detect the optimal install paths:
+if(HAVE_PYTHON)
+  if(WIN32 AND NOT CYGWIN)
+    set(_LIB_DIR ".")
+    set(_ARCH_DIR ".")
+
+  elseif(PYTHON_EXECUTABLE)
+    execute_process(
+      COMMAND ${PYTHON_EXECUTABLE}
+        -c "from distutils.sysconfig import get_python_lib; print(get_python_lib(False))"
+      OUTPUT_VARIABLE _LIB_DIR
+      OUTPUT_STRIP_TRAILING_WHITESPACE)
+    execute_process(
+      COMMAND ${PYTHON_EXECUTABLE}
+        -c "from distutils.sysconfig import get_python_lib; print(get_python_lib(True))"
+      OUTPUT_VARIABLE _ARCH_DIR
+      OUTPUT_STRIP_TRAILING_WHITESPACE)
+
+  else()
+    set(_LIB_DIR "")
+    set(_ARCH_DIR "")
+
+  endif()
+
+  execute_process(
+    COMMAND ${PYTHON_EXECUTABLE}
+      -c "from sysconfig import get_config_var as g; print((g('EXT_SUFFIX') or g('SO'))[:])"
+    OUTPUT_VARIABLE _EXT_SUFFIX
+    ERROR_QUIET
+    OUTPUT_STRIP_TRAILING_WHITESPACE)
```

Comment:
We should probably not call this if we're making a cross build.

Reply:
In a cross build, the Python executable won't be found and this won't get called. (This also works if you don't have a Python executable at all, and you only provide the headers.)

Reply:
Is CMake clever enough not to try to find things in the system when doing a cross build?  Then it's fine, I suppose (though I assume there's still a way to get at a host Python executable for running scripts like makewheel or so).

Reply:
The appropriate thing to do when cross-compiling using CMake is to define a "toolchain file" which contains info about where to find the cross buildroot, what special compile flags are needed, what compiler to use, and such a file usually contains this line:
`set(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM NEVER)`

...that causes CMake to look at the host when searching for programs (like Python; though I don't know if FindPython guarantees the interpreter is part of the same installation as the library, in which case it would not find it)

If you're feeling particularly hardcore you can use QEMU + binfmts on Linux to make non-native executables work as if they were native, then you can use `ONLY` instead of `NEVER` in the above. But I do want to make sure it works without having to do that.

Reply:
I might prefer a check for this just in case the user *hasn't* set this, but I don't feel strongly and you can resolve this as you please.

Reply:
I'll choose to wait and see if anyone's bitten by forgetting to set it before adding a check myself. I feel like the users who cross-compile will either use an existing toolchain file or (if they're setting up the environment themselves) they know what they're getting into and can debug it if they make a mistake.

But if someone does complain, we can revisit.